app/assets/javascripts/discourse/routes/topic-from-params.js.es6

```diff
@@ -76,5 +81,12 @@ export default Discourse.Route.extend({
           console.log("Could not view topic", e);
         }
       });
+  },
+
+  actions: {
+    willTransition(transition) {
+      const url = transition.router.generate(transition.from.name);
```

Comment:
Could we get the URL from `document.location` instead since it's not private APIs?